offset.go

```diff
@@ -0,0 +1,96 @@
+package vmap
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type (
+	// Duration is a time.Duration alias
+	Duration time.Duration
+
+	// Offset is a VAMP duration expressed as  hh:mm:ss
+	Offset struct {
+		timeDur  Duration
+		position OffsetPosition
+		percent  int
+	}
+
+	// OffsetPosition is for either passing start/end as the position of the ad
+	OffsetPosition string
+)
+
+const (
+	// OffsetPositionStart "start"
+	OffsetPositionStart OffsetPosition = "start"
+	// OffsetPositionEnd "end"
+	OffsetPositionEnd OffsetPosition = "end"
+)
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (off *Offset) MarshalText() ([]byte, error) {
+	if off.position == OffsetPositionStart || off.position == OffsetPositionEnd {
+		return []byte(off.position), nil
+	}
+
+	if off.percent != 0 {
+		return []byte(fmt.Sprintf("%d%%", off.percent)), nil
```

Comment:
I believe there should be something I'm not aware out which made you still return `nil` instead of an actual err. just curious

Reply:
I don't think this is an error since the offset is either percentage or time based?

Reply:
yeah this isn't an error. this whole struct's point is basically just circumventing strong typing and allowing an attribute to have several distinct types